### PR TITLE
fix(module): add back missing module export

### LINKED ISSUE #135: Cannot find module 'sanity:client' or its corresponding type declarations.
**Describe the bug**

I am following the instructions in [this article](https://www.sanity.io/plugins/sanity-astro) to build an astro project, but an error occurred during the build process.

This bug occurs during the build process and does not appear during the dev process.

Is this something I can solve on my own now? Or do I have to wait for you to fix it?

Thanks.

**To Reproduce**

Dependencies:

    "@astrojs/check": "^0.3.4",
    "@astrojs/react": "^3.0.9",
    "@sanity/astro": "^2.2.0",
    "@sanity/client": "^6.10.0",
    "@types/react": "^18.2.46",
    "@types/react-dom": "^18.2.18",
    "astro": "^4.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "sanity": "^3.23.4",
    "typescript": "^5.3.3"

astro.config.mjs:
```
import { defineConfig } from 'astro/config';
import { sanityIntegration } from "@sanity/astro";

export default defineConfig({
  integrations: [sanityIntegration({
    projectId: '***',
    dataset: '***',
    apiVersion: '2022-03-07',
    useCdn: false
  })],
  image: {
    domains: ["cdn.sanity.io"],
    remotePatterns: [{
      protocol: "https"
    }]
  }
});

```

Try to fetch data inside a component:

```
---
import { sanityClient } from "sanity:client";

export async function getStaticPaths() {
  const slugs = await sanityClient.fetch(`*[_type == 'book']{
  "slug": slug.current,
  "title": title
}`);
  const paths = await slugs.map((book: Book) => {
    return {
      params: {
        book: book.slug,
      },
      props: {title: book.title}
    };
  });
  return paths;
}
```

run `npm run build` will cause this error: 

 `error ts(2307): Cannot find module 'sanity:client' or its corresponding type declarations.`

**Expected behavior**

should build successfully

**Which versions of Sanity are you using?**

```
@sanity/cli (global)  3.23.4 (up to date)
@sanity/astro          2.2.0 (up to date)
sanity                3.23.4 (up to date)
```

**What operating system are you using?**

macOS 14.2.1

**Which versions of Node.js / npm are you running?**

18.18.0/10.1.0

### PR DESCRIPTION
**Pull request by b3nk3**
Hello team 🏄🏻 !

tldr;

This PR intends to fix the issue where importing from `sanity:client` throws an error, despite the correct triple-slash directive in `env.d.ts`. Should resolve #135.
CleanShot 2024-01-17 at 00 23 24

More info:

I've realised, that `/// <reference types="@sanity/astro/module" />` did not actually link to anywhere when doing a `⌘ + Click`. 

Upon this discovery, I've noticed, that on `v2.2.0` the export pointing to the `module.d.ts` file has been removed, leading to a reference to nowhere and the files contents not being imported.

I've forked the package and linked to it locally, validating that once the export is added back it works.

### CODE REVIEW

**Comment by narration-sd** [comment]
Ben, this is fully appreciated work you did, be sure!

I believe it's already covered on a safer way, in PR #137, which also includes @olach's important fix.

I've posted explanation points in [this fresh comment on the original issue](https://github.com/sanity-io/sanity-astro/issues/135#issuecomment-1894788093), so will keep things short here.

But again, appreciated work on something not so easy
Clive

**Comment by narration-sd** [comment]
With my original #137 now [closed](https://github.com/sanity-io/sanity-astro/pull/137#issuecomment-1898785828), I'd like to put my support behind approving this alternate PR, so that '@sanity/astro` will return to full and easy usefulness.

This is indeed a change of mind, a comfortable one due to progress [here](https://github.com/sanity-io/sanity-astro/issues/135#issuecomment-1898751690) in the issue discussion, with full regard to Ben's approach.

**Review by mirshko — APPROVED** [review]

**Comment by lokkeestudios** [comment]
Seems like this PR still needs 1 approving review. Would really appreciate this PR going through 😊 This error is driving me crazy

**Comment by olach** [comment]
@kmelve @stipsan could you or someone else on your team please have a look at this PR? There are also other stale issues and PRs that need to be handled. This official Sanity + Astro integration has been more or less broken for months.

**Review by stipsan — APPROVED** [review]
Thank you for the fix and apologies for the wait! 🙇

**Comment by stipsan** [comment]
Here you go: https://www.npmjs.com/package/@sanity/astro/v/2.2.1
